### ISSUE TITLE
Implement rand traits for Ratio ?

### ISSUE THREAD

**Issue by pwnorbitals** [rust-num/num#409]
I'm especially interested in the implementation of `Distribution<num::Ratio> for Standard` and the `UniformSampler` backend to use `gen_range()` on custom types that wrap `num::Ratio`

**Comment by cuviper** [comment]
Do you know how to actually achieve a uniform sample with a ratio? I think that's not easy, unless you just pick a specific denominator and then uniformly randomize the numerator. Is there a better way?

**Comment by clarfonthey** [comment]
I agree with cuviper -- I'm not 100% sure uniform randomness inside a range for `Ratio` is something that can be easily quantified. Most people will have their own idea of how the algorithm should work, and mathematically it's not really well-founded since all definitions of probability rely on real numbers, which can't easily be approximated with ratios. In the absolute case, if you wanted to generate a random `Ratio<BigInt>`, I don't think that it'd be desirable that there's a chance that the numerator and denominator can be arbitrarily large.

The only solution that makes sense to me is to just randomly generate a float and then convert that to a ratio to whatever precision you want. Which, can easily be accomplished by doing… just that.
